jax/lib/xla_bridge.py

```diff
@@ -292,6 +292,55 @@ def constant(builder, py_val, canonicalize_types=True):
   else:
     raise TypeError("No constant handler for type: {}".format(py_type))
 
+# HLO instructions optionally can be annotated to say how the output should be
+# spatially partitioned (represented in XLA as OpSharding protos, see
+# _sharding_to_proto). For array outputs, the annotation is either an int per
+# dimension specifying the number of ways that dimension divided (meaning the
+# total number of shards in the product), or None to indicate the array should
+# be replicated. Tuple outputs are represented as tuples thereof. XLA supports
+# arbitrary tuple nesting, but JAX only uses one level of tupling (and our type
```

Comment:
This wouldn't have to have the outer level of tupling if `tuple_args` shenanigans were pushed into PJRT, right?

Reply:
I also use it for the outputs tuple, which I think we'll always need to build directly. (We also build the inputs tuple ourselves now, not sure if there are plans to have PJRT rewrite the actual HLO parameters into a tuple.) Although I could probably use a custom_call instead of putting the annotation directly on the inputs/outputs, and then I wouldn't need tuples. I'm gonna keep this since it's already done though.